src/libraries/System.Diagnostics.TextWriterTraceListener/src/System.Diagnostics.TextWriterTraceListener.csproj

```diff
@@ -10,24 +10,6 @@
     <Compile Include="System\Diagnostics\DelimitedListTraceListener.cs" />
```

Comment:
This does not need to have OS specific build anymore.

`<TargetFrameworks>$(NetCoreAppCurrent)<TargetFrameworks>`

Reply:
Yeah, I tried that, but I got errors that I wasn't sure what to do with, and so decided to leave that for a follow-up, e.g.
```
C:\Users\stoub\.nuget\packages\microsoft.dotnet.build.tasks.targetframework.sdk\5.0.0-beta.20316.1\build\Microsoft.DotNet.Build.Tasks.TargetFramework.Sdk.targets(74,5): error : Not able to find a compatible supported target framework for net5.0 in Project System.Diagnostics.Process.csproj. The Supported Configurations are net5.0-Windows_NT, net5.0-FreeBSD, net5.0-Linux, net5.0-OSX, net5.0-iOS, net5.0-tvOS [D:\repos\runtime\src\libraries\System.Diagnostics.TraceSource\src\System.Diagnostics.TraceSource.csproj]
```

Reply:
Can you remove the reference to System.Diagnostics.Process?  Is its still used anywhere?  IIRC there was a case for Process.GetCurrentProcess().ProcessName which was unused in some cases.  Perhaps we can use a different pattern to obtain the name of the process, like AppDomain.CurrentDomain.FriendlyName 🤮 or add Environment.ProcessName.

Reply:
@ericstj, I can remove the Process reference, but then the error just becomes:
```
C:\Users\stoub\.nuget\packages\microsoft.dotnet.build.tasks.targetframework.sdk\5.0.0-beta.20316.1\build\Microsoft.DotNet.Build.Tasks.TargetFramework.Sdk.targets(74,5): error : Not able to find a compatible supported target framework for net5.0 in Project System.IO.FileSystem.csproj. The Supported Configurations are net5.0-Windows_NT, net5.0-Unix [D:\repos\runtime\src\libraries\System.Diagnostics.TraceSource\src\System.Diagnostics.TraceSource.csproj]
```
for FileSystem.

Reply:
It looks to me like you're talking about a different project?  This comment is on TextWriterTraceListener and the error mentions `System.Diagnostics.TraceSource.csproj` and I don't see any project references in TextWriterTraceListener.

In the case of TraceSource you might be able to get away with changing from a `ProjectReference` to a simple `Reference`.  ProjectReferences are only needed when you need to use a type that lives in System.Private.CoreLib at runtime (to avoid the duplicate type error from compiler).

Reply:
> It looks to me like you're talking about a different project? This comment is on TextWriterTraceListener and the error mentions System.Diagnostics.TraceSource.csproj and I don't see any project references in TextWriterTraceListener.

I was talking about TraceSource (the below "Same here" comment was applying to TraceSource as well, so I've just been using that as a guinea pig).  For TextWriterTraceListener, it does still use Process, to get the current process name.

> In the case of TraceSource you might be able to get away with changing from a ProjectReference to a simple Reference. 

The purpose of this PR is to add Environment.ProcessId.  I'm not going to hold it up right now fiddling with the csproj.  Please open an issue about consolidating the target frameworks if there isn't already one tracking that. Thanks!

Reply:
Ok, that's fine by me.  /cc @Anipik

Reply:
Great, thanks.

Reply:
I can follow up with this and see where we can leverage this api and remove cross targeting